apps/aecore/src/aec_block_key_candidate.erl

```diff
@@ -7,71 +7,87 @@
 -module(aec_block_key_candidate).
 
 -export([ create/3
+        , hc_create_hole/3
```

Comment:
We should avoid, if possible, to change the API of this module with HC specific things (unless we do it at compile time).

Look at the solution for `Height` below, make a callback in the consensus-module and take it that route?

Reply:
Yes, perhaps compile-time ifdefs are the way to go here. 
It would be a step towards an HC fork/branch of the repo.
Having a callback to determine if you are producing a hole feels convoluted.
It would require some extra make setup but perhaps not much more complicated than the config setup we need to build the complete docker.

Reply:
> Having a callback to determine if you are producing a hole feels convoluted.

That is what you get when your fitting the round peg in the square hole...